arch/sim/src/Makefile

```diff
@@ -195,8 +195,15 @@ endif
 
 # Override in Make.defs if linker is not 'ld'
 
-LDSTARTGROUP ?= --start-group
-LDENDGROUP ?= --end-group
+ifeq ($(HOSTOS),Darwin)
+  LDSTARTGROUP ?=
+  LDENDGROUP ?=
+  LDUNEXPORTSYMBOLS ?= -unexported_symbols_list $(HOSTOS)-names.dat
```

Comment:
If so, why we don't update other platform to use -unexported_symbols_list? to simplify the logic in Makefile.

Reply:
do gnu ld have the option? my copy of its documentation doesn't have it.

Reply:
Emm, unexported_symbols_list doesn't support on Linux, but after searching we can install binutils and use gobjcopy?

Reply:
as i mentioned in the commit logs, gnu binutils didn't actually work for me.
objcopy runs without errors, producing a corrupted object.

Reply:
Ok.